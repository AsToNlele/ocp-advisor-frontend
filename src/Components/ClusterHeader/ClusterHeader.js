import React from 'react';
import PropTypes from 'prop-types';

import {
  Grid,
  GridItem,
  Stack,
  StackItem,
  Title,
} from '@patternfly/react-core';
import { useIntl } from 'react-intl';
import Skeleton from '@redhat-cloud-services/frontend-components/Skeleton';

import messages from '../../Messages';
import { useGetClusterDisplayNameByIdQuery } from '../../Services/AccountManagementService';

const ClusterHeader = ({ uuid, lastSeen }) => {
  const intl = useIntl();
  const {
    data: displayName,
    isUninitialized,
    isLoading,
    isFetching,
  } = useGetClusterDisplayNameByIdQuery(uuid);

  return (
    <React.Fragment>
      <Grid md={12} hasGutter>
        <GridItem>
          {isUninitialized || isLoading || isFetching ? (
            <Skeleton size="sm" />
          ) : (
            <Title size="2xl" headingLevel="h1">
              {displayName || uuid}
            </Title>
          )}
        </GridItem>
        <GridItem>
          <Stack>
            <StackItem>
              <span>UUID: </span>
              <span>{uuid || intl.formatMessage(messages.unknown)}</span>
            </StackItem>
            <StackItem>
              <span>{intl.formatMessage(messages.lastSeen)}: </span>
              <span>{lastSeen || intl.formatMessage(messages.unknown)}</span>
            </StackItem>
          </Stack>
        </GridItem>
      </Grid>
    </React.Fragment>
  );
};

ClusterHeader.propTypes = {
  uuid: PropTypes.string.isRequired,
  lastSeen: PropTypes.string,
};

export default ClusterHeader;
