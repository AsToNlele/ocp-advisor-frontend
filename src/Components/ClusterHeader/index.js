import React from 'react';
import { routerParams } from '@redhat-cloud-services/frontend-components-utilities/RouterParams/RouterParams';

import { useGetClusterDisplayNameByIdQuery } from '../../Services/AccountManagementService';
import { useLazyGetClusterByIdQuery } from '../../Services/SmartProxy';
import { ClusterHeader } from './ClusterHeader';

export default routerParams(({ match }) => {
  const clusterId = match.params.clusterId;
  const displayName = useGetClusterDisplayNameByIdQuery(clusterId);
  const { data } = useLazyGetClusterByIdQuery(clusterId);

  return (
    <ClusterHeader
      clusterId={clusterId}
      displayName={displayName}
      lastSeen={data?.report?.lastSeen}
    />
  );
});
