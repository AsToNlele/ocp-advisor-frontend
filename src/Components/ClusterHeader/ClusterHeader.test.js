import { shallow } from 'enzyme';
import React from 'react';

import { ClusterHeader } from './ClusterHeader';

describe('<ClusterHeader /> test', () => {
  let props;

  beforeEach(() => {
    props = {
      clusterId: 'foobar',
      displayName: {
        isError: false,
        isUninitialized: false,
        isLoading: false,
        isFetching: false,
        isSuccess: true,
        data: 'Cluster with issues',
      },
      lastSeen: '2021-07-24T14:22:36.109Z',
    };
  });
  test('renders correctly', () => {
    const shallowed = shallow(<ClusterHeader {...props} />);
    expect(shallowed).toMatchSnapshot();
  });
  test('renders correctly when loading', () => {
    props.displayName.isFetching = true;
    props.displayName.isSuccess = false;
    const shallowed = shallow(<ClusterHeader {...props} />);
    expect(shallowed).toMatchSnapshot();
  });
  test('renders correctly without display name available', () => {
    props.displayName.data = undefined;
    const shallowed = shallow(<ClusterHeader {...props} />);
    expect(shallowed).toMatchSnapshot();
  });
  test('renders correctly without last seen available', () => {
    props.lastSeen = undefined;
    const shallowed = shallow(<ClusterHeader {...props} />);
    expect(shallowed).toMatchSnapshot();
  });
});
