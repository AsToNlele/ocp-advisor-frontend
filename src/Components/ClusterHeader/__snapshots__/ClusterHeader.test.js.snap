// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClusterHeader /> test renders correctly 1`] = `
<Fragment>
  <Grid
    hasGutter={true}
    md={12}
  >
    <GridItem>
      <Title
        headingLevel="h1"
        size="2xl"
      >
        Cluster with issues
      </Title>
    </GridItem>
    <GridItem>
      <Stack>
        <StackItem>
          <span>
            UUID: 
          </span>
          <span>
            foobar
          </span>
        </StackItem>
        <StackItem>
          <span>
            Last seen
            : 
          </span>
          <span>
            2021-07-24T14:22:36.109Z
          </span>
        </StackItem>
      </Stack>
    </GridItem>
  </Grid>
</Fragment>
`;

exports[`<ClusterHeader /> test renders correctly when loading 1`] = `
<Fragment>
  <Grid
    hasGutter={true}
    md={12}
  >
    <GridItem>
      <Skeleton
        isDark={false}
        size="sm"
      />
    </GridItem>
    <GridItem>
      <Stack>
        <StackItem>
          <span>
            UUID: 
          </span>
          <span>
            foobar
          </span>
        </StackItem>
        <StackItem>
          <span>
            Last seen
            : 
          </span>
          <span>
            2021-07-24T14:22:36.109Z
          </span>
        </StackItem>
      </Stack>
    </GridItem>
  </Grid>
</Fragment>
`;

exports[`<ClusterHeader /> test renders correctly without display name available 1`] = `
<Fragment>
  <Grid
    hasGutter={true}
    md={12}
  >
    <GridItem>
      <Title
        headingLevel="h1"
        size="2xl"
      >
        foobar
      </Title>
    </GridItem>
    <GridItem>
      <Stack>
        <StackItem>
          <span>
            UUID: 
          </span>
          <span>
            foobar
          </span>
        </StackItem>
        <StackItem>
          <span>
            Last seen
            : 
          </span>
          <span>
            2021-07-24T14:22:36.109Z
          </span>
        </StackItem>
      </Stack>
    </GridItem>
  </Grid>
</Fragment>
`;

exports[`<ClusterHeader /> test renders correctly without last seen available 1`] = `
<Fragment>
  <Grid
    hasGutter={true}
    md={12}
  >
    <GridItem>
      <Title
        headingLevel="h1"
        size="2xl"
      >
        Cluster with issues
      </Title>
    </GridItem>
    <GridItem>
      <Stack>
        <StackItem>
          <span>
            UUID: 
          </span>
          <span>
            foobar
          </span>
        </StackItem>
      </Stack>
    </GridItem>
  </Grid>
</Fragment>
`;
